aries_cloudagent/connections/models/diddoc/diddoc.py

```diff
@@ -113,9 +140,9 @@ def set(self, item: Union[Service, PublicKey]) -> "DIDDoc":
         """
 
         if isinstance(item, Service):
-            self.service[item.id] = item
+            self.service.append(item)
```

Comment:
Being stored as a dictionary does not strictly reflect the DID spec data model but it does make it easier to retrieve a service block out of the data structure. I think changing this to a list will make it less usable overall.

Reply:
I agree with you, however, for a good deserialize/serialize using marshmallow, the data structure from the JSON and from the class implementation must be the same.

Reply:
Ah, I see the challenge here now. We could perform a transformation during `post_load` and `pre_dump`, couldn't we?